fastly/resource_fastly_service_acl_entries_v1.go

```diff
@@ -0,0 +1,282 @@
+							Type:        schema.TypeString,
+							Description: "An IP address that is the focus for the ACL",
+							Required:    true,
+						},
+						"subnet": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "An optional subnet mask applied to the IP address",
+						},
+						"negated": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     false,
+							Description: "A boolean that will negate the match if true",
+						},
+						"comment": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "A personal freeform descriptive note",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceServiceAclEntriesV1Create(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*FastlyClient).conn
+
+	serviceID := d.Get("service_id").(string)
+	aclID := d.Get("acl_id").(string)
+	entries := d.Get("entry").(*schema.Set)
+
+	var batchACLEntries = []*gofastly.BatchACLEntry{}
+
+	for _, vRaw := range entries.List() {
+		val := vRaw.(map[string]interface{})
+
+		batchACLEntries = append(batchACLEntries, &gofastly.BatchACLEntry{
+			Operation: gofastly.CreateBatchOperation,
+			IP:        val["ip"].(string),
+			Subnet:    val["subnet"].(string),
+			Negated:   val["negated"].(bool),
+			Comment:   val["comment"].(string),
+		})
+	}
+
+	// Process the batch operations
+	err := executeBatchACLOperations(conn, serviceID, aclID, batchACLEntries)
+	if err != nil {
+		return fmt.Errorf("Error creating ACL entries: service %s, ACL %s, %s", serviceID, aclID, err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", serviceID, aclID))
+	return resourceServiceAclEntriesV1Read(d, meta)
+}
+
+func resourceServiceAclEntriesV1Read(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*FastlyClient).conn
+
+	serviceID := d.Get("service_id").(string)
+	aclID := d.Get("acl_id").(string)
+
+	aclEntries, err := conn.ListACLEntries(&gofastly.ListACLEntriesInput{
+		Service: serviceID,
+		ACL:     aclID,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("entry", flattenAclEntries(aclEntries))
+	return nil
+}
+
+func resourceServiceAclEntriesV1Update(d *schema.ResourceData, meta interface{}) error {
+
+	conn := meta.(*FastlyClient).conn
+
+	serviceID := d.Get("service_id").(string)
+	aclID := d.Get("acl_id").(string)
+
+	var batchACLEntries = []*gofastly.BatchACLEntry{}
+
+	if d.HasChange("entry") {
+
+		oe, ne := d.GetChange("entry")
+		if oe == nil {
+			oe = new(schema.Set)
+		}
+		if ne == nil {
+			ne = new(schema.Set)
+		}
+
+		oes := oe.(*schema.Set)
+		nes := ne.(*schema.Set)
+
+		removeEntries := oes.Difference(nes).List()
+		addEntries := nes.Difference(oes).List()
+
+		// DELETE old ACL entry
+		for _, vRaw := range removeEntries {
+			val := vRaw.(map[string]interface{})
+
+			batchACLEntries = append(batchACLEntries, &gofastly.BatchACLEntry{
+				Operation: gofastly.DeleteBatchOperation,
+				ID:        val["id"].(string),
+			})
+		}
+
```

Comment:
The function doesn't handle replaces (UpdateBatchOperation). This causes twice as many calls as necessary. The edge dictionary implementation does handle replaces vie UpdateBatchOperation.

Reply:
I'm assuming because this is a `TypeSet` you won't have replaces/updates. @trentrosenbaum can confirm?